archethic/contracts/factory.exs

```diff
@@ -94,11 +94,34 @@ export fun get_chargeable_htlc(end_time, user_address, pool_address, secret_hash
       Contract.set_code ""
     end
 
-    condition triggered_by: transaction, on: refund(), as: [
-      timestamp: timestamp >= #{end_time}
+    condition triggered_by: transaction, on: refund(evm_contract), as: [
```

Comment:
Here you don't need to pass the evm contract in parameter since we already get it in the provision action.